lib/repository_cache.rb

```diff
@@ -0,0 +1,25 @@
+# Interface to the Redis-backed cache store used by the Repository model
+class RepositoryCache
+  attr_reader :namespace
+
+  def initialize(namespace, backend = Rails.cache)
+    @namespace = namespace
+    @backend = backend
+  end
+
+  def cache_key(type)
+    "#{type}:#{namespace}"
+  end
+
+  def expire(key)
+    backend.delete(cache_key(key))
+  end
+
+  def fetch(key, &block)
+    backend.fetch(cache_key(key), &block)
+  end
+
+  private
+
+  attr_reader :backend
```

Comment:
I wanted the `attr_reader` for backend but it should be private, which traditionally goes at the bottom. I think you can do `private attr_reader :backend` all on one line and at the top, but that's a bit unusual.